backend/services/event.py

```diff
@@ -55,6 +57,75 @@ def __init__(
         self._permission = permission
         self._user_svc = user_svc
 
+    def get_paginated_events(
+        self,
+        pagination_params: EventPaginationParams,
+        subject: User | None = None,
+    ) -> Paginated[EventDetails]:
+        """List Events.
+
+        Parameters:
+            pagination_params: The pagination parameters.
+
+        Returns:
+            Paginated[Event]: The paginated list of events.
+        """
+
+        statement = select(EventEntity)
+        length_statement = select(func.count()).select_from(EventEntity)
+        if pagination_params.range_start != "":
+            range_start = pagination_params.range_start
+            range_end = pagination_params.range_end
+            criteria = and_(
+                EventEntity.time
+                >= datetime.strptime(range_start, "%d/%m/%Y, %H:%M:%S"),
+                EventEntity.time <= datetime.strptime(range_end, "%d/%m/%Y, %H:%M:%S"),
+            )
+            statement = statement.where(criteria)
+            length_statement = length_statement.where(criteria)
+
+        if pagination_params.filter != "":
+            query = pagination_params.filter
+
+            criteria = or_(
+                EventEntity.name.ilike(f"%{query}%"),
+                EventEntity.description.ilike(f"%{query}%"),
+                exists().where(
+                    OrganizationEntity.id == EventEntity.organization_id,
+                    OrganizationEntity.name.ilike(f"%{query}%"),
+                ),
+                exists().where(
+                    OrganizationEntity.id == EventEntity.organization_id,
+                    OrganizationEntity.slug.ilike(f"%{query}%"),
+                ),
+            )
+            statement = statement.where(criteria)
+            length_statement = length_statement.where(criteria)
+
+        offset = pagination_params.page * pagination_params.page_size
+        limit = pagination_params.page_size
+
+        if pagination_params.order_by != "":
+            if pagination_params.ascending == "false":
+                statement = statement.order_by(
+                    getattr(EventEntity, pagination_params.order_by).desc()
+                )
+            else:
+                statement = statement.order_by(
+                    getattr(EventEntity, pagination_params.order_by)
+                )

```

Comment:
Nit: Ternary operator might be able to make this more concise, but not required.